src/actions/breakpoints/breakpointPositions.js

```diff
@@ -44,10 +47,18 @@ function convertToList(results, sourceId) {
 export function setBreakpointPositions(location: SourceLocation) {
```

Comment:
Should we split this in two to keep the caching very clear, e.g.

```
const requests = new Map();
export function setBreakpointPositions({ sourceId }: SourceLocation) {
  return async (thunkArgs: ThunkArgs) => {
    let inProgress = requests.get(sourceId);
    if (!inProgress) {
      inProgress = setBreakpointPositionsPromise(thunkArgs, sourceId).finally(() => {
        requests.delete(sourceId);
      });
      requests.set(sourceId, inProgress);
    }
    return inProgress;
  };
}
async function setBreakpointPositionsPromise(({ getState, dispatch, client, sourceMaps }: ThunkArgs, sourceId: SourceID) {
  // ...
}
```

I think this helps make the general behavior a lot clearer.

Reply:
oh nice. I was looking for a clean way to do this